pkgs/applications/window-managers/notion/notion-xft_nixos.diff

```diff
@@ -0,0 +1,662 @@
++                        (based_on ? based_on->cgrp.bg : black));
+                         
+-    padinh=(based_on ? based_on->cgrp.pad : DE_WHITE(rootwin));
++    padinh=(based_on ? based_on->cgrp.pad : white);
+     
+     de_get_colour_(rootwin, &(cg->pad), tab, "padding_colour", 
+                    (bgset ? cg->bg : padinh), padinh);
+diff -ur notion-3-2015061300/de/style.c notion-3-2015061300-PATCHED/de/style.c
+--- notion-3-2015061300/de/style.c	2015-06-13 10:37:26.000000000 -0400
++++ notion-3-2015061300-PATCHED/de/style.c	2015-07-13 10:37:23.128296929 -0400
+@@ -70,10 +70,17 @@
+     /*gcv.function=GXclear;*/
+     gcv.stipple=stipple_pixmap;
+     gcvmask=GCFillStyle|GCStipple/*|GCFunction*/;
++#ifndef XFT
+     if(style->font!=NULL && style->font->fontstruct!=NULL){
+         gcv.font=style->font->fontstruct->fid;
+         gcvmask|=GCFont;
+     }
++#else /* XFT */
++//    if(style->font!=NULL){
++//        gcv.font=style->font;
++//        gcvmask|=GCFont;
++//    }
++#endif /* XFT */
+ 
+     style->stipple_gc=XCreateGC(dpy, root, gcvmask, &gcv);
+     XCopyGC(dpy, style->normal_gc, 
+@@ -201,6 +208,14 @@
+ 
+ bool destyle_init(DEStyle *style, WRootWin *rootwin, const char *name)
+ {
++    DEColour black, white;
++#ifdef XFT
++    de_alloc_colour(rootwin, &black, "black");
++    de_alloc_colour(rootwin, &white, "white");
++#else
++    black=DE_BLACK(rootwin);
++    white=DE_WHITE(rootwin);
++#endif /* XFT */
+     if(!gr_stylespec_load(&style->spec, name))
+         return FALSE;
+     
+@@ -223,11 +238,11 @@
+     style->textalign=DEALIGN_CENTER;
+ 
+     style->cgrp_alloced=FALSE;
+-    style->cgrp.bg=DE_BLACK(rootwin);
+-    style->cgrp.pad=DE_BLACK(rootwin);
+-    style->cgrp.fg=DE_WHITE(rootwin);
+-    style->cgrp.hl=DE_WHITE(rootwin);
+-    style->cgrp.sh=DE_WHITE(rootwin);
++    style->cgrp.bg=black;
++    style->cgrp.pad=black;
++    style->cgrp.fg=white;
++    style->cgrp.hl=white;
++    style->cgrp.sh=white;
+     gr_stylespec_init(&style->cgrp.spec);
+     
+     style->font=NULL;
+diff -ur notion-3-2015061300/de/style.h notion-3-2015061300-PATCHED/de/style.h
+--- notion-3-2015061300/de/style.h	2015-06-13 10:37:26.000000000 -0400
++++ notion-3-2015061300-PATCHED/de/style.h	2015-07-13 10:37:23.130296815 -0400
+@@ -80,6 +80,7 @@
+     Pixmap tag_pixmap;
+     int tag_pixmap_w;
+     int tag_pixmap_h;
++    int xft_style;
+     
+     DEStyle *next, *prev;
+ };
+diff -ur notion-3-2015061300/system-autodetect.mk notion-3-2015061300-PATCHED/system-autodetect.mk
+--- notion-3-2015061300/system-autodetect.mk	2015-06-13 10:37:26.000000000 -0400
++++ notion-3-2015061300-PATCHED/system-autodetect.mk	2015-07-13 10:38:14.921332013 -0400
+@@ -103,6 +103,19 @@
+ 
+ 
+ ##
++## Xft support
++##
++
++USE_XFT=1
++
++ifeq ($(USE_XFT),1)
++EXTRA_INCLUDES += `pkg-config xft --cflags`
++EXTRA_LIBS += `pkg-config xft --libs`
++DEFINES += -DXFT -DCF_FALLBACK_FONT_NAME=\"${fallbackFont}\"
```

Comment:
Are you sure this works? You don't pass this `fallbackFont` variable anywhere through the derivation. I think you missed a `inherit fallbackFont` in the drv.